### PR TITLE
retries for group_manifests plugin

### PR DESCRIPTION
**Pull request by rcerven**
Signed-off-by: Robert Cerven <rcerven@redhat.com>

### CODE REVIEW

**Comment by vrutkovs** [comment]
Need retries for:
* `check_output` of manifest-tool runs
* replace retries code in `get_manifest_digests`
* no need to `response.raise_for_status()` in `get_worker_manifest` if it 404 - otherwise it would take 15 mins to realize that Pulp doesn't support schema 2

**Comment by twaugh** [comment]
It might be easier to add `check_output` retries once we have a solution for `git clone` as it is a similar problem.

Not sure whta you mean by replacing retries code in `get_manifest_digests`.

Why isn't `response.raise_for_status()` needed for 404? This plugin does not interact with Pulp, only with the Docker distribution registry.

**Comment by vrutkovs** [comment]
>Not sure whta you mean by replacing retries code in get_manifest_digests.

Right, this code was already updated

>Why isn't response.raise_for_status() needed for 404? This plugin does not interact with Pulp, only with the Docker distribution registry.

I was thinking registry (and pulp) would return 404 when a different schema is tried, but its actually 400, so nevermind

File: atomic_reactor/plugins/post_group_manifests.py [hunk 1]
```diff
@@ -164,7 +165,8 @@ def get_worker_manifest(self, worker_data):
 
                 url = '{0}/v2/{1}/manifests/{2}'.format(registry, repo, digest)
                 self.log.debug("attempting get from %s", url)
-                response = requests.get(url, **kwargs)
+                response = session.get(url, **kwargs)
+                response.raise_for_status()
```
Comment (lcarva) [inline atomic_reactor/plugins/post_group_manifests.py]: @mlangsdorf , we noticed that for this particular `get` API call the status code is not checked.
We're adding it now. Can you confirm this was not a deliberate decision?

**Comment by lcarva** [comment]
Not updating release notes draft because this is bugfix to an unreleased feature.

**Pull request merged**